internal/entitlement/postgresadapter/ent/schema/entitlement.go

```diff
@@ -25,11 +26,19 @@ func (Entitlement) Mixin() []ent.Mixin {
 
 func (Entitlement) Fields() []ent.Field {
 	return []ent.Field{
+		field.Enum("entitlement_type").Values(entitlement.EntitlementType("").StrValues()...).Immutable(),
 		field.String("feature_id").Immutable().SchemaType(map[string]string{
 			dialect.Postgres: "char(26)",
 		}),
 		field.String("subject_key").Immutable(),
-		field.Time("measure_usage_from").Immutable(),
+		field.Time("measure_usage_from").Optional().Nillable().Immutable(),
+		field.Float("issue_after_reset").Optional().Nillable().Immutable(),
+		field.Bool("is_soft_limit").Optional().Nillable().Immutable(),
+		field.String("config").Optional().Nillable().SchemaType(map[string]string{
```

Comment:
Based on this comment: https://github.com/openmeterio/openmeter/blob/55fb3fcc153f7d1db926043d31ecd0ded26ad521/api/api.gen.go#L390

Config should a json object, but as far as I see:
- In the schema it's a text object
- In the code here https://github.com/openmeterio/openmeter/blob/96b5deeb7369c9e314fd5fb942da4d3c9fc87ff6/internal/entitlement/static/static_entitlement.go#L10

What's the idea here? If we expect it to be json, then we should validate it at some point, right?

Reply:
My thinking was I don't know what format we'll want it to be. The api description says it should be json. We'll probably be safer if we enforce it to be json for now. 
I can see a rational behind not limiting it to being json and I didn't want to really tackle that, that's why i pushed it off.
I'll just enforce it to be json.

Reply:
I think this https://github.com/openmeterio/openmeter/pull/1042/files#diff-6c08e65eef0374988297d041bd50f87b385613a7e478c3812cb5312cc3e1f1e3R1915 should be an object typed member. It feels strange that we are embedding JSON string into a json response.

Reply:
enforced to be json